features/nanostack/sal-stack-nanostack/source/Service_Libs/hmac/hmac_sha1.c

```diff
@@ -44,16 +44,12 @@ int8_t hmac_sha1_calc(const uint8_t *key, uint16_t key_len, const uint8_t *data,
     }
 #endif
 
-    const mbedtls_md_type_t md_type = MBEDTLS_MD_SHA1;
-    const mbedtls_md_info_t *md_info = mbedtls_md_info_from_type(md_type);
-    if (md_info == NULL) {
-        return -1;
-    }
-
     mbedtls_md_context_t ctx;
 
     mbedtls_md_init(&ctx);
-    if (mbedtls_md_setup(&ctx, md_info, 1) != 0) {
+
+    /* mbedtls_md_setup will fail with MBEDTLS_ERR_MD_BAD_INPUT_DATA if mbedtls_md_info_from_type returns NULL */
+    if (mbedtls_md_setup(&ctx, mbedtls_md_info_from_type(MBEDTLS_MD_SHA1), 1) != 0) {
```

Comment:
In commit " Enhanced Hmac handling, needed for MbedTLS optimisations", it's not clear to me why this change is actually needed as the commit message says so. It's a refactoring that looks to preserve the previous behavior. Please explain in the commit message why the change is needed, if it is actually needed, or change the commit message to explain this is a refactoring.

Reply:
@jarvte Can you comment for this?

Reply:
That's not really specific enough. What about the upcoming changes requires this? I am skeptical the changes are needed; convince me in the commit message.

Reply:
removed that commit completely

Reply:
OK